flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/ReconciliationUtils.java

```diff
@@ -118,6 +122,29 @@ public static void updateForReconciliationError(FlinkDeployment flinkApp, String
         return updateControl.rescheduleAfter(rescheduleAfter.toMillis());
     }
 
+    public static boolean isUpgradeModeChangedToLastStateAndHADisabledPreviously(
+            FlinkDeployment flinkApp) {
+        final FlinkDeploymentSpec lastReconciledSpec =
+                Preconditions.checkNotNull(
+                        flinkApp.getStatus().getReconciliationStatus().getLastReconciledSpec());
+        if (lastReconciledSpec.getJob() == null || flinkApp.getSpec().getJob() == null) {
+            return false;
+        }
+        if (lastReconciledSpec.getFlinkConfiguration() == null
+                || lastReconciledSpec.getFlinkConfiguration().isEmpty()) {
+            return false;
```

Comment:
If upgrade mode is last state these will never be null due to validation, so we can remove the check